### PR TITLE
Allow to override reporter in Server

### PR DESCRIPTION
**Pull request by ertrzyiks**
Related to #41

I don't see any specs for this file yet, I tested it manually on my machine. Is it enough or you want some specs for this too?

### CODE REVIEW

**Comment by ai** [comment]
Unfortunately, we need tests for it.

Server tests looks tricky, but it is not hard. Create a new server with custom reporter at `test/servers/` and check their output by snapshots in `server.test.js`.

**Comment by ertrzyiks** [comment]
This was also my impression that specs are needed here, I was asking just in case. I will add them tomorrow 👍

**Comment by ai** [comment]
Also maybe we should pass `reporter` in `options` object? 

I used the second option in `BaseServer` only because `reporter` was some sort private option. If we make in public, we can add it to `options` in `BaseServer` too.

File: server.js [hunk 1]
```diff
@@ -141,8 +141,11 @@ class Server extends BaseServer {
 
   constructor (options) {
     if (!options) options = { }
-    options.reporter = options.reporter || 'human'
-    const reporter = createReporter(options)
+
+    if (typeof options.reporter !== 'function') {
+      options.reporter = options.reporter || 'human'
+      options.reporter = createReporter(options)
```
Comment (ertrzyiks) [inline server.js]: I don't like it to be honest, maybe something like?

```
createReporter(type, options)
```

**Comment by ertrzyiks** [comment]
@ai Actually, it was easier than expected. There were specs for this class already, I missed them at first glance.

What do you think?

**Comment by ai** [comment]
Great PR. I especially like your idea to use `reporter` for functions too.

**Pull request merged**